### ISSUE TITLE
Usage of `warning.warn` in pya

### ISSUE THREAD

**Issue by aleneum** [interactive-sonification/pya#12]
I see that @wiccy46 uses `warning.warn` quite frequently and I wonder if that is the right choice. `warning.warn` is commonly used to signal *usage* issues (using deprecated methods, passing incomplete/wrong parameters to a function etc.) whereas `logging.warning` is used to document/log processes. `warning.warn` is by default only emitted once and thus not useful to track/debug issues. `warning` can also be configured to handle warnings as exceptions.

tl;dr: If  users should change their code, use `warning.warn`; for everything else use `logging.warning` (or better `logger_instance.warning`)

**Comment by wiccy46** [comment]
I check through the codes. Now warn is used to inform user where they should change the code. We used to have a few places where warn should be replaced by raise so that correct behaviour can be expected.
